### ISSUE TITLE
Initialization error on startup of Homebridge

### ISSUE THREAD

**Issue by monteroman** [homebridge-xiaomi-roborock-vacuum/homebridge-xiaomi-roborock-vacuum#185]
My plugin started throwing this error this morning.

Rosie] Initializing XiaomiRoborockVacuum accessory...
[5/11/2020, 6:29:11 AM] TypeError: this.services.fan.setPrimaryService is not a function
    at XiaomiRoborockVacuum.initialiseServices (/usr/lib/node_modules/homebridge-xiaomi-roborock-vacuum/index.js:144:23)
    at new XiaomiRoborockVacuum (/usr/lib/node_modules/homebridge-xiaomi-roborock-vacuum/index.js:120:10)
    at Server._loadAccessories (/usr/lib/node_modules/homebridge/lib/server.js:308:29)
    at Server.run (/usr/lib/node_modules/homebridge/lib/server.js:91:38)
    at module.exports (/usr/lib/node_modules/homebridge/lib/cli.js:59:10)
    at Object.<anonymous> (/usr/lib/node_modules/homebridge/bin/homebridge:17:22)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1043:10)
    at internal/main/run_main_module.js:17:11

Is there something I can try to work around this?

**Comment by afharo** [comment]
Are you running the latest version of `homebridge`?

**Comment by monteroman** [comment]
Yes.

**Comment by afharo** [comment]
`1.0.4`? https://www.npmjs.com/package/homebridge

I am running that one and I have no issues.

**Comment by afharo** [comment]
Anyway, I published `0.8.1` to maintain backwards compatibility

**Comment by monteroman** [comment]
The version I'm running is 1.0.4 on UnRaid.   My UnRaid system took the docker system offline this weekend for its weekly backup and brought it back online, but the plugin started throwing an error which prevented homebridge from starting up.

**Comment by afharo** [comment]
Weird... I'm hearing from others that after upgrading `homebridge` to `1.0.4` it worked. Anyway, `0.8.1` should offer backwards compatibility and do not crash

**Comment by monteroman** [comment]
Ohh... I see what happened.   I had to make a change to my docker settings because I added a new plugin that I wanted to make sure stayed in the configuration.   After it redeployed and restarted the docker app, it reverted to an older version of Homebridge.  I was able to re-update to 1.0.4 within the gui after it initialized.   Thanks for putting the backwards compatibility routines in.   I'll reach out to the maintainer of the docker image to see if they can update their package so it is at 1.0.4.

Thank you for making the change so quickly!
